File: java/gestionStock_springBoot/src/main/java/com/edugroupe/gestionstock_springboot/entity/Categorie.java
package com.edugroupe.gestionstock_springboot.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import lombok.*;

import javax.persistence.*;
import java.io.Serializable;
import java.util.List;

@Entity
@Table(name = "categorie")
@NoArgsConstructor
@AllArgsConstructor
@Setter
@Getter
@ToString
@EqualsAndHashCode
public class Categorie implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    private String nom;
    @JsonIgnore
    @OneToMany(cascade = CascadeType.ALL, fetch = FetchType.LAZY, mappedBy = "categorie")
    private List<Produit> produits;





}

File: gescom_angular/src/app/produits/categorie.ts
export class Categorie {

  private _id: number = 0;
  private _nom: string = '';


  get id(): number {
    return this._id;
  }

  set id(value: number) {
    this._id = value;
  }

  get nom(): string {
    return this._nom;
  }

  set nom(value: string) {
    this._nom = value;
  }

  toString() :String{
    return "{id:" + this.id +" , nome:"+ this.nom +"}";
  }

}
